src/test/java/uk/gov/pay/api/utils/mocks/ConnectorMockClient.java

```diff
@@ -61,44 +67,35 @@
 
     public ConnectorMockClient(int port, String baseUrl) {
         super(port, baseUrl);
     }
-
-    private String buildChargeResponse(long amount, String chargeId, PaymentState state, String returnUrl, String description,
-                                       String reference, String email, String paymentProvider, String gatewayTransactionId, String createdDate,
-                                       SupportedLanguage language, boolean delayedCapture, Long corporateCardSurcharge, Long totalAmount, RefundSummary refundSummary, SettlementSummary settlementSummary, CardDetails cardDetails,
-                                       ImmutableMap<?, ?>... links) {
+    
+    private String buildChargeResponse(CreateChargeResponseFromConnector responseFromConnector) {
         PaymentSingleResultBuilder resultBuilder = aSuccessfulSinglePayment()
-                .withChargeId(chargeId)
-                .withAmount(amount)
-                .withMatchingReference(reference)
-                .withEmail(email)
-                .withDescription(description)
-                .withState(state)
-                .withReturnUrl(returnUrl)
-                .withCreatedDate(createdDate)
-                .withLanguage(language)
-                .withPaymentProvider(paymentProvider)
-                .withDelayedCapture(delayedCapture)
-                .withLinks(asList(links))
-                .withRefundSummary(refundSummary)
-                .withSettlementSummary(settlementSummary)
-                .withCardDetails(cardDetails);
-
-        if (gatewayTransactionId != null) {
-            resultBuilder.withGatewayTransactionId(gatewayTransactionId);
-        }
-
-        if (corporateCardSurcharge != null) {
-            resultBuilder.withCorporateCardSurcharge(corporateCardSurcharge);
-        }
-
-        if (totalAmount != null) {
-            resultBuilder.withTotalAmount(totalAmount);
-        }
-
+                .withChargeId(responseFromConnector.chargeId)
+                .withAmount(responseFromConnector.amount)
+                .withMatchingReference(responseFromConnector.reference)
+                .withEmail(responseFromConnector.email)
+                .withDescription(responseFromConnector.description)
+                .withState(responseFromConnector.state)
+                .withReturnUrl(responseFromConnector.returnUrl)
+                .withCreatedDate(responseFromConnector.createdDate)
+                .withLanguage(responseFromConnector.language)
+                .withPaymentProvider(responseFromConnector.paymentProvider)
+                .withDelayedCapture(responseFromConnector.delayedCapture)
+                .withLinks(responseFromConnector.links)
+                .withSettlementSummary(responseFromConnector.settlementSummary)
+                .withCardDetails(responseFromConnector.cardDetails);
+        
+        if (responseFromConnector.refundSummary != null) resultBuilder.withRefundSummary(responseFromConnector.refundSummary);
```

Comment:
I thought our style included always using `{}` with control statements, although I'm not sure I could find it written down to point to it...

Reply:
can't remember if our style mandated this but i'll move the `resultBuilder.withRefundSummary...` down to a new line

Reply:
I doubt it does. I agree with your other comment, as long as its clear and obvious to all is the important thing.